task-bht_bold.psyexp

```diff
@@ -213,8 +213,8 @@
         <Param val="True" valType="bool" updates="constant" name="syncScreenRefresh"/>
       </KeyboardComponent>
       <CodeComponent name="code_7" plugin="None">
-        <Param val="import socket&amp;#10;def send_message(message, addr=&quot;localhost&quot;, port=2023):&amp;#10;    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)&amp;#10;    client_socket.connect((addr, port))&amp;#10;    client_socket.sendall(message)&amp;#10;    client_socket.close()&amp;#10;" valType="extendedCode" updates="constant" name="Before Experiment"/>
-        <Param val="import * as socket from 'socket';&amp;#10;function send_message(message, addr = &quot;localhost&quot;, port = 2023) {&amp;#10;    var client_socket;&amp;#10;    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM);&amp;#10;    client_socket.connect([addr, port]);&amp;#10;    client_socket.sendall(message);&amp;#10;    client_socket.close();&amp;#10;}&amp;#10;" valType="extendedCode" updates="constant" name="Before JS Experiment"/>
+        <Param val="import socket&amp;#10;def send_message(message, addr=&quot;localhost&quot;, port=2023):&amp;#10;    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)&amp;#10;    client_socket.connect((addr, port))&amp;#10;    client_socket.sendall(message)&amp;#10;    client_socket.close()&amp;#10;folder_path = os.path.dirname(os.path.abspath(__file__))+ os.sep + 'data/'&amp;#10;ename = 'breath_holding_task_'+data.getDateStr(format=&quot;%Y-%m-%d&quot;)&amp;#10;trial = 0&amp;#10;for file in os.listdir(folder_path):&amp;#10;    if ename in file and '.log' in file:&amp;#10;        trial += 1" valType="extendedCode" updates="constant" name="Before Experiment"/>
```

Comment:
The question is whether this is executed before the modal dialog asks for the trial number.

Reply:
But I trust you have tested it, so please go ahead and merge whenever you feel comfortable merging.

Reply:
The trial number is implemented by default as the Python variable trial. But it's a good point, I hadn't considered the scenario where someone might enter a random number instead of directly clicking 'OK.' I'll make sure to double-check before merging. Thanks!